File: dictionary_decrypt.c
/* Brian Lee
 * 3620101
 * CMPSC 177
 * Brute-force dictionary attack
 * Compile using: gcc -lcrypt dictionary_decrypt.c -o dictionary_decrypt
 */

#define _XOPEN_SOURCE 

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

int main(int argc, char *argv[])
{
    if(argc != 3) {
    printf("Usage is %s hash salt\n", argv[0]);
    return 1;
  }
    //argv[1] is target hash, argv[2] is salt
    FILE *dictionary;
    char buffer[1000];
    char* hash;
    char* toHash;
    //int i = 1;
    
    dictionary = fopen("words.txt", "r");
    if(!dictionary)
    {
        return 1;
    }
    while(fgets(buffer, sizeof(buffer), dictionary) != NULL)
    {
        toHash = strtok(buffer, "\n");
        hash = crypt(toHash, argv[2]);
	//printf("Iteration: %s\n", i);
        if(strcmp(hash, argv[1]) == 0)
        {
            printf("Password: %s\n", toHash);
            return 0;
        }
	else
	  {
	    printf("Nope\n");
	  }
	//i++;
    }
    
    printf("No password found\n");
    
    return 0;
}

File: dictionary_decrypt.py
'''
    Brian Lee
    3620101
    CMPSC 177
    Dictionary Brute-force attack
'''

import sys
import crypt

if len(sys.argv) != 3:
    print "Usage: {} hash salt".format(sys.argv[0])
else:
    with open("words.txt", "r") as dictionary:
        passwords = dictionary.read().splitlines()
    output = open("hash.txt", "wb")
    
    hash1 = sys.argv[1]
    salt = sys.argv[2]
    found = False
    
    for password in passwords:
        e_pass = crypt.crypt(password, salt)
	toPrint = e_pass + '\n'
        output.write(toPrint)
        if(e_pass == hash1):
            print "Password: {}".format(password)
            found = True
            break
    output.close();
    
    if(found == False):
        print "Password not found.\n"
